test/e2e/logforwarding/splunk/forward_to_splunk_test.go

```diff
@@ -0,0 +1,160 @@
+	"github.com/openshift/cluster-logging-operator/test/client"
+	framework "github.com/openshift/cluster-logging-operator/test/framework/e2e"
+	"github.com/openshift/cluster-logging-operator/test/helpers"
+	"github.com/openshift/cluster-logging-operator/test/helpers/oc"
+	"github.com/openshift/cluster-logging-operator/test/helpers/types"
+	"github.com/openshift/cluster-logging-operator/test/runtime"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sync/errgroup"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strings"
+	"testing"
+	"time"
+)
+
+const (
+	splunkReceiverName = "splunk-receiver"
+	message            = "To be, or not to be, that is the question"
+	logGenerator       = "log-generator"
+)
+
+var (
+	e2e    = framework.NewE2ETestFramework()
+	secret = corev1.Secret{
+
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: corev1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "vector-splunk-secret",
+			Namespace: constants.OpenshiftNS,
+		},
+		Type: "Opaque",
+		Data: map[string][]byte{
+			"hecToken": framework.HecToken,
+		},
+	}
+	spec = loggingv1.ClusterLogForwarderSpec{
+		Outputs: []loggingv1.OutputSpec{{
+			Name: splunkReceiverName,
+			Type: loggingv1.OutputTypeSplunk,
+			URL:  framework.SplunkEndpoint.String(),
+			OutputTypeSpec: loggingv1.OutputTypeSpec{
+				Splunk: &loggingv1.Splunk{},
+			},
+			Secret: &loggingv1.OutputSecretSpec{
+				Name: "vector-splunk-secret",
+			},
+		}},
+		Pipelines: []loggingv1.PipelineSpec{
+			{
+				Name:       "test-app",
+				InputRefs:  []string{loggingv1.InputNameApplication, loggingv1.InputNameInfrastructure},
+				OutputRefs: []string{splunkReceiverName},
+			},
+		},
+	}
+)
+
+func TestLogForwardingToSplunkWithVector(t *testing.T) {
+	_, err := e2e.KubeClient.CoreV1().Secrets(constants.OpenshiftNS).Create(context.TODO(), &secret, metav1.CreateOptions{})
+	if err != nil {
+		if apierrors.IsAlreadyExists(err) {
+			sOpts := metav1.UpdateOptions{}
+			_, err := e2e.KubeClient.CoreV1().Secrets(constants.OpenshiftNS).Update(context.TODO(), &secret, sOpts)
+			if err != nil {
+				Fail(fmt.Sprintf("Unable to craete secret for Vector: %v", err))
+			}
+		} else {
+			fmt.Println(err)
+			Fail(fmt.Sprintf("Unable to craete secret for Vector: %v", err))
+		}
+	}
+	if _, err := e2e.DeploySplunk(); err != nil {
+		Fail(fmt.Sprintf("Unable to deploy splunk receiver: %v", err))
+	}
+	cl := runtime.NewClusterLogging()
+	cl.Spec.Collection.Type = loggingv1.LogCollectionTypeVector
+	cl.Spec.Collection.CollectorSpec = loggingv1.CollectorSpec{}
+	clf := runtime.NewClusterLogForwarder()
+	clf.Spec = spec
+	testLogForwardingToSplunk(t, cl, clf)
+
+}
+
+func testLogForwardingToSplunk(t *testing.T, cl *loggingv1.ClusterLogging, clf *loggingv1.ClusterLogForwarder) {
+	c := client.ForTest(t)
+	defer framework.RunCleanupScript()
```

Comment:
Can we defer "framework.Cleanup" which may also run this cleanup script?

Reply:
done